ror-tools-core/src/main/scala/tech/beshu/ror/tools/core/patches/internal/modifiers/bytecodeJars/DummyAuthenticationInAuthenticationChain.scala

```diff
@@ -0,0 +1,297 @@
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.tools.core.patches.internal.modifiers.bytecodeJars
+
+import just.semver.SemVer
+import org.objectweb.asm.*
+import tech.beshu.ror.tools.core.patches.internal.modifiers.BytecodeJarModifier
+import tech.beshu.ror.tools.core.utils.EsUtil.*
+
+import java.io.{File, InputStream}
+
+private[patches] class DummyAuthenticationInAuthenticationChain(esVersion: SemVer) extends BytecodeJarModifier {
+
+  override def apply(jar: File): Unit = {
+    modifyFileInJar(
+      jar = jar,
+      filePathString = "org/elasticsearch/xpack/security/authc/AuthenticatorChain.class",
+      processFileContent = putDummyAuthenticationInAuthenticationChain
+    )
+  }
+
+  private def putDummyAuthenticationInAuthenticationChain(moduleInputStream: InputStream) = {
+    val reader = new ClassReader(moduleInputStream)
+    val writer = new ClassWriter(reader, 0)
+    reader.accept(new EsClassVisitor(writer), 0)
+    writer.toByteArray
+  }
+
+  private class EsClassVisitor(writer: ClassWriter)
+    extends ClassVisitor(Opcodes.ASM9, writer) {
+
+    override def visitMethod(access: Int,
+                             name: String,
+                             descriptor: String,
+                             signature: String,
+                             exceptions: Array[String]): MethodVisitor = {
+      name match {
+        case "doAuthenticate" =>
+          esVersion match {
+            case v if v >= es8120 =>
+              new SetXpackUserAsAuthenticatedUserForEsGreaterOrEqual8120(super.visitMethod(access, name, descriptor, signature, exceptions))
+            case v if v >= es890 =>
+              new SetXpackUserAsAuthenticatedUserForEsGreaterOrEqual890x(super.visitMethod(access, name, descriptor, signature, exceptions))
+            case v if v >= es870 =>
+              new SetXpackUserAsAuthenticatedUserForEsGreaterOrEqual870(super.visitMethod(access, name, descriptor, signature, exceptions))
+            case v if v >= es820 =>
+              new SetXpackUserAsAuthenticatedUserForEsGreaterOrEqual820(super.visitMethod(access, name, descriptor, signature, exceptions))
+            case v if v >= es71713 =>
+              new SetXpackUserAsAuthenticatedUserForEsGreaterOrEqual71713(super.visitMethod(access, name, descriptor, signature, exceptions))
+            case v if v >= es7160 =>
+              new SetXpackUserAsAuthenticatedUserForEsGreaterOrEqual71713(super.visitMethod(access, name, descriptor, signature, exceptions))
```

Comment:
Is the first case needed? Or do the es7160 versions need a dedicated method visitor?